src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/RegexRustTests.cs

```diff
@@ -0,0 +1,443 @@
+                yield return new object[] { engine, @".", "\xD4\xC2\x65\x2B\x0E\xFE", new[] { (0, 1), (1, 2), (2, 3), (3, 4), (4, 5), (5, 6) } };
+                yield return new object[] { engine, @"${2}\u00E4", "\xD4\xC2\x65\x2B\x0E\xFE", new ValueTuple<int, int>[] { } };
+                yield return new object[] { engine, @"\u2603", "\u2603", new[] { (0, 1) } };
+                yield return new object[] { engine, @"\u2603+", "\u2603", new[] { (0, 1) } };
+                yield return new object[] { engine, @"(?i)\u2603+", "\u2603", new[] { (0, 1) } };
+                yield return new object[] { engine, @"[\u2603\u2160]+", "\u2603", new[] { (0, 1) } };
+                yield return new object[] { engine, @"(?i)\u0394", "\u03B4", new[] { (0, 1) } };
+                yield return new object[] { engine, @"\p{Lu}+", "\u039B\u0398\u0393\u0394\u03B1", new[] { (0, 4) } };
+                yield return new object[] { engine, @"(?i)\p{Lu}+", "\u039B\u0398\u0393\u0394\u03B1", new[] { (0, 5) } };
+                yield return new object[] { engine, @"\p{L}+", "\u039B\u0398\u0393\u0394\u03B1", new[] { (0, 5) } };
+                yield return new object[] { engine, @"\p{Ll}+", "\u039B\u0398\u0393\u0394\u03B1", new[] { (4, 5) } };
+                yield return new object[] { engine, @"\w+", "d\u03B4d", new[] { (0, 3) } };
+                yield return new object[] { engine, @"\w+", "\u2961", new ValueTuple<int, int>[] { } };
+                yield return new object[] { engine, @"\W+", "\u2961", new[] { (0, 1) } };
+                yield return new object[] { engine, @"\d+", "1\u0968\u09699", new[] { (0, 4) } };
+                yield return new object[] { engine, @"\d+", "\u2161", new ValueTuple<int, int>[] { } };
+                yield return new object[] { engine, @"\D+", "\u2161", new[] { (0, 1) } };
+                yield return new object[] { engine, @"\s+", "\u1680", new[] { (0, 1) } };
+                yield return new object[] { engine, @"\s+", "\u2603", new ValueTuple<int, int>[] { } };
+                yield return new object[] { engine, @"\S+", "\u2603", new[] { (0, 1) } };
+                yield return new object[] { engine, @"\d\b", "6\u03B4", new ValueTuple<int, int>[] { } };
+                yield return new object[] { engine, @"\d\b", "6\u1680", new[] { (0, 1) } };
+                yield return new object[] { engine, @"\d\B", "6\u03B4", new[] { (0, 1) } };
+                yield return new object[] { engine, @"\d\B", "6\u1680", new ValueTuple<int, int>[] { } };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(MatchStartAndEndPositionsTestData))]
+        public async Task MatchStartAndEndPositions(RegexEngine engine, string pattern, string input, IEnumerable<(int, int)>? matchBoundaries)
+        {
+            Regex regex = await RegexHelpers.GetRegexAsync(engine, pattern);
+            Match match = regex.Match(input);
+            if (matchBoundaries == null)
```

Comment:
Where is it null?

Reply:
It's not, good point. I had initially made this parameter nullable and would pass null for no matches, but I switched that so I'll remove this if condition.